client/dashboard/app/queries/site-logs.ts

```diff
@@ -0,0 +1,51 @@
+import { queryOptions, keepPreviousData } from '@tanstack/react-query';
+import {
+	fetchSiteLogs,
+	SiteLogsAPIResponse,
+	SiteLogsData,
+	SiteLogsParams,
+	PHPLog,
+	ServerLog,
+} from '../../data/site-logs';
+
+export const siteLogsQuery = ( siteId: number, params: SiteLogsParams ) =>
+	queryOptions< SiteLogsAPIResponse, unknown, SiteLogsData >( {
+		queryKey: [
+			'site',
+			siteId,
+			'logs',
+			params.logType,
+			params.start,
+			params.end,
+			JSON.stringify( params.filter ),
+			params.sortOrder,
+			params.pageSize,
+			params.pageIndex,
+		],
+		queryFn: () =>
+			fetchSiteLogs(
+				siteId,
+				params.logType,
+				params.start,
+				params.end,
+				params.filter,
+				params.sortOrder,
+				params.pageSize,
+				params.pageIndex
+			),
+		placeholderData: keepPreviousData,
+		enabled: !! siteId && params.start <= params.end,
+		staleTime: Infinity, // The logs within a specified time range never change.
+		select: ( data: SiteLogsAPIResponse ) => ( {
```

Comment:
Similar to [this comment](https://github.com/Automattic/wp-calypso/pull/105118/files#r2262172589), since we're using `queryOptions()` we shouldn't explicitly define the function parameter's type here. It will pick up the correct type automatically.

```suggestion
		select: ( data ) => ( {
```

Reply:
Makes sense, included that in df820d253a8faef125c9bf96762aeba0f102fec2